### ISSUE TITLE
Migrating to Raft storage backend

### ISSUE THREAD

**Issue by chrissng** [hashicorp/vault#7198]
**Describe the bug**
I have been exploring Vault 1.2.0-rc1 as I am interested in the Raft storage backend. It makes setting up and managing the storage  simpler. I attempted to migrate to this new backend but encountered errors.

However, migration does not seem possible. According to the migration docs (https://www.vaultproject.io/docs/commands/operator/migrate.html), the storage destination should not be initialized. However, a vault cluster with raft storage backend cannot be created as nodes cannot join the cluster when vault is still sealed. Unsealing vault requires initialization.

**To Reproduce**
Prepare migration script:
```hcl
storage_source "gcs" {
  bucket = "chrissng-testing-vault"
}

storage_destination "raft" {
  path = "/home/ubuntu/Downloads/vault/raft_state"
  node_id = "raft_node_1"
}
```

No init approach - Steps to reproduce the behavior:
1. Create empty directory where raft state will be store
2. Run `vault operator migrate -config=migrate.hcl`
3. Get error `Error migrating: error writing entry: raft storage backend is not initialized`

Init destination approach - Steps to reproduce the behavior:
1. Initialize and unseal Vault cluster with Raft backend
2. Run `vault operator migrate -config=migrate.hcl`
3. Get error `Error migrating: error mounting 'storage_destination': failed to create fsm: timeout`

**Expected behavior**
Keys copied over to new backend.

**Environment:**
* Vault Server Version (retrieve with `vault status`): v1.2.0-rc1
* Vault CLI Version (retrieve with `vault version`): v1.2.0-rc1
* Server Operating System/Architecture: Ubuntu 18.04.2 LTS

Vault server configuration file(s): 

Vault standalone with GCS backend:
```hcl
storage "gcs" {
  bucket = "chrissng-testing-vault"
}

listener "tcp" {
  address = "127.0.0.1:8200"
  tls_disable = 1
}

disable_mlock = true
```

Vault cluster server 1:
```hcl
storage "raft" {
  path = "/home/ubuntu/Downloads/vault/raft_state"
  node_id = "raft_node_1"
}

listener "tcp" {
  address = "127.0.0.1:8300"
  tls_disable = 1
}

api_addr = "http://127.0.0.1:8300"
cluster_addr = "http://127.0.0.1:8301"
disable_mlock = true
```

Vault cluster server 2:
```hcl
storage "raft" {
  path = "/home/ubuntu/Downloads/vault/raft_state2"
  node_id = "raft_node_2"
}

listener "tcp" {
  address = "127.0.0.1:8400"
  tls_disable = 1
}

api_addr = "http://127.0.0.1:8400"
cluster_addr = "http://127.0.0.1:8401"
disable_mlock = true
```
Vault cluster server 3:
```hcl
storage "raft" {
  path = "/home/ubuntu/Downloads/vault/raft_state3"
  node_id = "raft_node_3"
}

listener "tcp" {
  address = "127.0.0.1:8500"
  tls_disable = 1
}

api_addr = "http://127.0.0.1:8500"
cluster_addr = "http://127.0.0.1:8501"
disable_mlock = true
```

**Additional context**

**Comment by chrissng** [comment]
@briankassouf and @vishalnayak As you are the code and documentation authots, if you can provide some inputs, that will help! Thank you!

**Comment by briankassouf** [comment]
Thanks for the detailed report! This is a known improvement we are planning on tackling in the near term. The migrator will need to be adapted to know how to bootstrap a raft cluster without needed to initialize it first.

**Comment by briankassouf** [comment]
fixed by #7207
